### ISSUE TITLE
Lambda resolver for GraphQL API: return `@model` objects

### ISSUE THREAD

**Issue by ataibarkai** [aws-amplify/amplify-cli#7907]
**Which Category is your question related to?**
api + function
(GraphQL API / Lambda Function resolver)

**Amplify CLI Version** 
5.2.1

**What AWS Services are you utilizing?**
Practically everything supported by amplify, and a few others.

**Provide additional details e.g. code snippets**
I would like to create a custom query (i.e. using a lambda function) for an array of objects defined using the  `@model` directive. 

e.g.
```
customQuery(arg: String): [Post]
```

where `Post` is a `@model`-defined type.

If all I want to return is IDs of `Post`s, this is trivial.
But I would like to return actual `Post` objects, allowing the client to query for nested fields on each `Post` using the standard resolvers auto-generated for  `@model` objects.

What's the simplest way to achieve this?

**Comment by jhockett** [comment]
Hi @ataibarkai, I don't think a custom resolver is necessary. Is there a reason not to use something like the following?:

```gql
type Query {
    customQuery: Post
}
```

**Comment by ataibarkai** [comment]
Hi @jhockett , thank you for your reply.

My use-case must use a lambda function, because the `[Post]` list returned by the query is fully custom, and in fact depends on a 3rd-party service which the function queries against. The `Post`s do not correspond to the results of a simple DynamoDB query.

However after some experimenting I was able to answer my own question:
Once you have the relevant IDs, you simply need to query for the objects and return the 'flat' payload of the relevant dynamoDB `read` query.
`@connection` and `@function` fields on the response requiring subsequent (recursive) DB queries would be correctly resolved by the AppSync engine, automatically.

The pieces to implement this are all here right now. However significant performance gains can be had by making a very small (1-line) tweak to the `@function` directive. I just submitted a feature request to that effect: https://github.com/aws-amplify/amplify-cli/issues/7925
